src/asset-management/EulerV2Manager.sol

```diff
@@ -305,17 +261,64 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
                                 ADDITIONAL REWARDS
     //////////////////////////////////////////////////////////////////////////*/
 
-    function claimRewardForMarket(address aMarket, address aReward)
+    function claimRewards(
+        IDistributor aDistributor,
+        address[] calldata aUsers,
+        address[] calldata aTokens,
+        uint256[] calldata aAmounts,
+        bytes32[][] calldata aProofs
+    ) external onlyGuardianOrOwner {
+        aDistributor.claim(aUsers, aTokens, aAmounts, aProofs);
+
+        for (uint256 i = 0; i < aTokens.length; ++i) {
+            SafeTransferLib.safeTransfer(
+                aTokens[i],
+                msg.sender,
+                // the amounts specified in the argument might not be the actual amounts disimbursed by the distributor,
+                // due to the possibility of having done the claim previously thus it is necessary to use balanceOf
+                // to transfer the correct amount
+                IERC20(aTokens[i]).balanceOf(address(this))
+            );
+        }
+    }
+
+    /// @dev The guardian or owner would first call `claimRewards` and sell it for the underlying token.
+    /// The asset manager pulls the assets, deposits it to the vault, and distribute the proceeds in the form of ERC4626
+    /// shares to the pairs.
+    /// Due to integer arithmetic the last pair of the array will get one or two more shares, so as to maintain the
+    /// invariant that the sum of shares for all pair+token equals the totalShares.
+    function distributeRewardForPairs(IERC20 aAsset, uint256 aAmount, IAssetManagedPair[] calldata aPairs)
         external
         onlyGuardianOrOwner
-        returns (uint256 rClaimed)
+        nonReentrant
```

Comment:
This is the only function marked non reentrant why does this need it and not the others?

Reply:
actually it's this function plus the internal `_adjustManagement` that have the reentrancy guard. 

The reason is that only these two function change the state, so we have to be take more precautions. 

The rest the external functions are either view or are admin functions that don't modify the important states (`shares` and `totalShares`)

Reply:
Sounds good, nice